juju/client/connection.py

```diff
@@ -8,15 +8,106 @@
 from concurrent.futures import CancelledError
 from http.client import HTTPSConnection
 
-import macaroonbakery.httpbakery as httpbakery
 import macaroonbakery.bakery as bakery
+import macaroonbakery.httpbakery as httpbakery
 import websockets
 from juju import errors, tag, utils
 from juju.client import client
 from juju.utils import IdQueue
 
 log = logging.getLogger('juju.client.connection')
 
+client_facades = {
+    'Action': { 'version': 2 },
```

Comment:
don't these need to be lists of supported versions (even if all the current versions are only length 1) ?